daemon.go

```diff
@@ -101,6 +105,13 @@ func NewDaemon(
 		opts = append(opts, libp2p.Routing(d.DHTRoutingFactory(dhtOpts)))
 	}
 
+	if bandwidthMetricsEnabled {
+		d.bandwidth_metrics = metrics.NewBandwidthCounter()
+		opts = append(opts, libp2p.BandwidthReporter(d.bandwidth_metrics))
+	} else {
+		d.bandwidth_metrics = nil
```

Comment:
nit: `d.bandwidth_metrics` are `nil` be default anyway.
ignore this comment if you wish